pkg/logs/input/file/tailer.go

```diff
@@ -231,7 +241,10 @@ func (t *Tailer) forwardMessages() {
 		origin.Identifier = identifier
 		origin.Offset = strconv.FormatInt(offset, 10)
 		origin.SetTags(append(t.tags, t.tagProvider.GetTags()...))
-		t.outputChan <- message.NewMessage(output.Content, origin, output.Status)
+		select {
+		case t.outputChan <- message.NewMessage(output.Content, origin, output.Status):
+		case <-t.forwardContext.Done():
```

Comment:
As discuss IRL, I would had a comment here to explain why we don't send the message to the output channel or don't directly return